src/pip/_internal/pep425tags.py

```diff
@@ -298,11 +235,16 @@ def _mac_platforms(arch):
     match = _osx_arch_pat.match(arch)
     if match:
         name, major, minor, actual_arch = match.groups()
-        tpl = '{}_{}_%i_%s'.format(name, major)
-        arches = []
-        for m in reversed(range(int(minor) + 1)):
-            for a in get_darwin_arches(int(major), m, actual_arch):
-                arches.append(tpl % (m, a))
+        mac_version = (int(major), int(minor))
+        arches = [
+            # Since we have always only checked that the platform starts
+            # with "macosx", for backwards-compatibility we extract the
+            # actual prefix provided by the user in case they provided
+            # something like "macosxcustom_". It may be good to remove
+            # this as undocumented or deprecate it in the future.
+            '{}_{}'.format(name, arch[len('macosx_'):])
+            for arch in mac_platforms(mac_version, actual_arch)
```

Comment:
The `arch` might be returned in a different order in some cases:
```
>>> from pip._internal.pep425tags import get_darwin_arches
>>> get_darwin_arches(10,5, 'ppc')
['ppc', 'fat', 'fat32', 'universal']
```
vs
```
>>> from packaging.tags import mac_platforms
>>> list(mac_platforms((10,5), 'ppc'))
['macosx_10_5_ppc', 'macosx_10_5_fat32', 'macosx_10_5_fat', 'macosx_10_5_universal', 'macosx_10_4_ppc', 'macosx_10_4_fat32', 'macosx_10_4_fat', 'macosx_10_4_universal', 'macosx_10_3_ppc', 'macosx_10_3_fat32', 'macosx_10_3_fat', 'macosx_10_3_universal', 'macosx_10_2_ppc', 'macosx_10_2_fat32', 'macosx_10_2_fat', 'macosx_10_2_universal', 'macosx_10_1_ppc', 'macosx_10_1_fat32', 'macosx_10_1_fat', 'macosx_10_1_universal', 'macosx_10_0_ppc', 'macosx_10_0_fat32', 'macosx_10_0_fat', 'macosx_10_0_universal']
```
(so `ppc', 'fat32', 'fat', 'universal'`)

A changelog might be a good idea to help macosx users know that something might have changed.

Reply:
I prefer to have a blanket statement that we improved tag calculation by switching to `packaging.tags`. OK to do that in #7354?

Reply:
> OK to do that in #7354?

Sure